src/order/category/DecLinOrder.lean

```diff
@@ -0,0 +1,77 @@
+/- Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+Introduce
+* Preorder     : the category of preorders and monotone functions.
+* PartialOrder : the category of partial orders and monotone functions.
+* LinOrder     : the category of linear orders and monotone functions.
+* DecLinOrder  : the category of decidable linear orders and monotone functions.
+
+-/
+
+import category_theory.concrete_category
+import order.basic
+
+universes u v
+
+open category_theory
+
+/-- The category of preorders and monotone maps. -/
+@[reducible]
+def Preorder : Type (u+1) := bundled preorder
+
+namespace Preorder
+
+def of (X : Type u) [preorder X] : Preorder := bundled.of X
+
+instance : unbundled_hom @monotone :=
+⟨@monotone_id, (λ _ _ _ _ _ _ _ _ m₁ m₂, by exactI monotone_comp m₂ m₁)⟩
+
+instance (G : Preorder) : preorder G := G.str
+
+example : large_category Preorder.{u} := infer_instance
+
+end Preorder
+
+/-- The category of partial orders and monotone maps. -/
+@[reducible]
+def PartialOrder : Type (u+1) := induced_category (bundled.map partial_order.to_preorder.{u})
```

Comment:
In the last few commits I pursued this, and I think it turns out quite nicely!

Reply:
@semorrison It seems that `naturality_bundled` was the only lemma outside of `concrete_category/` formulated for `bundled` instead of `concrete_category`. Thanks for writing `naturality_bundled'`! I copied it to cones, renamed to `naturality_concrete`, and deleted the `_bundled` version. Feel free to amend/fix/revert.

Reply:
We've now gone this route.